### DISCUSSION TITLE
Help identifying endpoint id for smart meter

### DISCUSSION THREAD

**Discussion by AStoker** [bemasher/rtlamr#304] [Q&A]
I'm trying to identify the endpoint for my meter. It's a Itron Gen 5 (Model: NIC 531-0601-13). I'm getting SCM+ messages with endpoints that don't seem to match anything on my meter. Searching the internet for help has only lead me to 5 year old questions on reddit :D I'm not sure if the stickers for FCC Ids have changed, or what's going on. Could somebody help me identify how to parse the endpoint ID?
BTW, this is what my output looks like
```
{Time:2024-07-03T16:17:51.726 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130418068 Consumption:     85258 Tamper:0x0800 PacketCRC:0x9DF1}}
{Time:2024-07-03T16:19:26.137 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130833899 Consumption:     49946 Tamper:0x0B00 PacketCRC:0x6A5C}}
{Time:2024-07-03T16:19:37.631 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130993539 Consumption:     57534 Tamper:0x0B00 PacketCRC:0x8177}}
{Time:2024-07-03T16:20:37.622 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130993539 Consumption:     57534 Tamper:0x0B00 PacketCRC:0x8177}}
{Time:2024-07-03T16:21:34.817 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130128871 Consumption:     77434 Tamper:0x0100 PacketCRC:0x8E03}}
{Time:2024-07-03T16:21:52.617 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130993539 Consumption:     57534 Tamper:0x0B00 PacketCRC:0x8177}}
{Time:2024-07-03T16:22:22.602 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130993539 Consumption:     57534 Tamper:0x0B00 PacketCRC:0x8177}}
{Time:2024-07-03T16:22:41.108 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130833899 Consumption:     49946 Tamper:0x0B00 PacketCRC:0x6A5C}}
{Time:2024-07-03T16:22:51.709 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130418068 Consumption:     85258 Tamper:0x0800 PacketCRC:0x9DF1}}
{Time:2024-07-03T16:23:06.715 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130418068 Consumption:     85258 Tamper:0x0800 PacketCRC:0x9DF1}}
{Time:2024-07-03T16:23:22.604 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130993539 Consumption:     57534 Tamper:0x0B00 PacketCRC:0x8177}}
{Time:2024-07-03T16:24:41.113 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130833899 Consumption:     49946 Tamper:0x0B00 PacketCRC:0x6A5C}}
{Time:2024-07-03T16:25:11.114 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130833899 Consumption:     49946 Tamper:0x0B00 PacketCRC:0x6A5C}}
{Time:2024-07-03T16:25:19.810 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130128871 Consumption:     77434 Tamper:0x0100 PacketCRC:0x8E03}}
{Time:2024-07-03T16:25:22.769 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130993539 Consumption:     57534 Tamper:0x0B00 PacketCRC:0x8177}}
{Time:2024-07-03T16:26:11.110 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130833899 Consumption:     49946 Tamper:0x0B00 PacketCRC:0x6A5C}}
{Time:2024-07-03T16:26:22.609 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130993539 Consumption:     57534 Tamper:0x0B00 PacketCRC:0x8177}}
{Time:2024-07-03T16:26:52.630 SCM+:{ProtocolID:0x1E EndpointType:0xFC EndpointID: 130993539 Consumption:     57534 Tamper:0x0B00 PacketCRC:0x8177}}
```

![IMG_4250](https://github.com/bemasher/rtlamr/assets/2907279/cb83fe2b-ecf9-45f4-b4db-9749949b2a75)

**Comment by bemasher** [comment]
Doesn't look like you've got a compatible meter. The FCC report has internal photos with a Silver Spring Networks IC, and the FCC label doesn't have an ID in the expected format for ERT.

**Reply by AStoker** [reply]
What would all those messages I'm seeing be then? All the neighbors have the same kind of meter

**Reply by AStoker** [reply]
Not that it adds much credibility, but it looks like at least on the community spreadsheet doc, somebody posted this one as working. Wish I knew for certain though :D

**Comment by Nils154** [comment]
For me the easiest was to identify the actual reading. I see a bunch of gas meters and water meters but no electrical with my setup.--Nils Gokemeijer (Sent from iPhone)On Jul 4, 2024, at 12:44 PM, Andrew Stoker ***@***.***> wrote:﻿
Not that it adds much credibility, but it looks like at least on the community spreadsheet doc, somebody posted this one as working. Wish I knew for certain though :D

**Comment by bbobrian** [comment]
Hey @AStoker I'm also near Columbia, SC (Chapin) and trying to read my Dominion meter as well. I get a bunch of different EndpointIDs starting with 130 just as you are, and they seem to be the gas meters. Check yours and I bet one of the IDs matches.

No luck so far with the electric meter. I may try to test some different frequencies and see if anything comes up. If I do find something I'll report back here.

**Reply by jcgoette** [reply]
Is your gas meter an ERG-7000-001 (FCC ID: EWQ500GA)? If so, any specific rtlamr setup to capture these? :)

**Reply by bbobrian** [reply]
That is my exact meter. Try `rtl_amr -msgtype=scm+`  and you should see messages within a minute or so. I pick up 3-4 different meters with the NooElec Nano 2 Plus and the telescoping antenna it came with.

**Comment by Doxpuppr** [comment]
Just got a SDR and was trying to get my HA instance set up with RTLAMR, and also in the Columbia area with a Dominion meter. Did y'all ever figure it out/get anything working to read your electric usage?
Thanks in advance!

**Reply by bbobrian** [reply]
Unfortunately, no. I actually went to revisit this a few weeks ago to at least get the gas meter data into HA, but was no longer able to get any readings at all.

As far as Dominion, my next thought was to try to access the API used in their "Energy Analyzer" interface. It's using a system called Bidgely, which I found a few projects trying to integrate with. That's as far as I've gotten.

**Comment by brownleeb** [comment]
I'm using an IR phototransistor to monitor the IR LEDs in the optical connector.  I think its the LED on the left.  On my meter, it flashes once every two W/hr.  A previous version of a meter had the same optical connector, but flashed every 12 W/hr.  I just had to watch it for a while to figure which was right.  I use an arduino as a A/D converter and have some filters programmed in to smooth out the signal.  I cant remember the name of the filter, but basically it just ignores any transition between (for example) 1 and 4 volts.  It assumes a high once the voltage crosses above 4v, and doesnt assume low until it crosses below 1v.  There is a lot of noise on my wire, its 20+ ft long, and sometimes flickers of sunlight cause jitter.
They used to read the meter manually using the optical coupler, but have since prgrammed it into the RF mesh, so I just cover the LED lens with electrical tape and no one ever disturbs it.  Just takes some sample analysis to find the right way to handle it.
